### PR TITLE
Issue 1 - Request

### LINKED ISSUE #1: Request implementation

### PR DESCRIPTION
**Pull request by jefim**
Closing #1 

## Summary by CodeRabbit

- **New Features**
  - Introduced a workflow for building and testing Microsoft Dynamics 365 requests.
  - Added task definition for making requests to Microsoft Dynamics 365.

- **Bug Fixes**
  - Added necessary `.gitignore` entries for `.idea` and `.env.local` files.

- **Documentation**
  - Provided a README with installation, build, and testing instructions.
  - Included Apache License 2.0 terms and conditions.

- **Refactor**
  - Updated project to target .NET 6.0 and included necessary package references and configurations.

- **Style**
  - Applied suppressions for StyleCop warnings to align with Frends documentation and coding guidelines.

### CODE REVIEW

File: Frends.MicrosoftDynamics365.Request/Frends.MicrosoftDynamics365.Request/Definitions/Result.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
+﻿namespace Frends.MicrosoftDynamics365.Request.Definitions;
+
+/// <summary>
+/// Contains properties of the return object.
+/// </summary>
+public class Result
+{
+    internal Result()
```
Comment (MatteoDelOmbra) [inline Frends.MicrosoftDynamics365.Request/Frends.MicrosoftDynamics365.Request/Definitions/Result.cs:8]: can be deleted

File: Frends.MicrosoftDynamics365.Request/Frends.MicrosoftDynamics365.Request.Tests/UnitTests.cs [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    {
+        var input = GetInput("bad_request");
+        var options = GetOptions(true);
+
+        var exception = Assert.ThrowsAsync<Exception>(async () =>
+            await MicrosoftDynamics365.Request(input, options, CancellationToken.None));
+        Assert.That(exception.Message, Does.Contain("Error retrieving bad_request. Status code: NotFound"));
+    }
+
+    private Input GetInput(string path, Method method = Method.GET, string payload = null)
```
Comment (MatteoDelOmbra) [inline Frends.MicrosoftDynamics365.Request/Frends.MicrosoftDynamics365.Request.Tests/UnitTests.cs]: could be neater with using "=>" expression, but it's only semantics :D
    private Input GetInput(string path, Method method = Method.GET, string payload = null) => new(){...};

File: Frends.MicrosoftDynamics365.Request/Frends.MicrosoftDynamics365.Request.Tests/UnitTests.cs [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+            Path = path,
+            ClientId = ClientId,
+            TenantId = TenantId,
+            ClientSecret = ClientSecret,
+            Dynamics365Url = Dynamics365Url,
+            Method = method,
+        };
+    }
+
+    private Options GetOptions(bool throwExceptionOnErrorResponse = false)
```
Comment (MatteoDelOmbra) [inline Frends.MicrosoftDynamics365.Request/Frends.MicrosoftDynamics365.Request.Tests/UnitTests.cs]: same as for input

**Review by MatteoDelOmbra — COMMENTED** [review]
Great job! Btw - tests that we are doing are not UnitTest imo, so maybe we should change name of UnitTests.cs file/class to just Tests or IntegrationTests?

File: Frends.MicrosoftDynamics365.Request/Frends.MicrosoftDynamics365.Request/Definitions/Result.cs (hunk 1 above)
Comment (jefim) [reply Frends.MicrosoftDynamics365.Request/Frends.MicrosoftDynamics365.Request/Definitions/Result.cs:8]: internal result is that this in not instantiatable from outside the task

Comment (MatteoDelOmbra) [reply Frends.MicrosoftDynamics365.Request/Frends.MicrosoftDynamics365.Request/Definitions/Result.cs:8]: yes but we don't need this empty constructor at all

**Review by MatteoDelOmbra — APPROVED** [review]

**Pull request merged**